calcom/cal.com
# Detect users browser locale for time format 12/24 hours

## What does this PR do?
This PR introduces a new util ~~timeFormat~~ -> renamed to detectBrowserTimeFormat to detect the browser locale of the user and format the dates accordingly.

I need to integrate it everywhere, started on BookingPage.
We respect the current implementation that lets the user switch the am/pm settings, but aut-detect it's browser locale to set it by default in the booking page and rest of the site (where is not configurable)

Fixes #1378

In places like payment page, sucess, meeting ended, etc I've left the new browserLocale instead of using useState (since it wasn't using the clock is24h util either on those pages)

Let me know what do you think and if I missed any time I need to fix please!

## Type of change

## How should this be tested?
Go to create a new booking, select a date and a time. Make sure the time is in the format of your browser locale. If you set 'en-US' as your default in chrome, you should see it with am/pm attached. 'es-ES' should show a 24h format without it.

You can test by reordering your top-language in chrome settings. Remember to delete the localStorage timeOptions.is24hClock setting so it takes the browser defaults instead
- [x] Test  'en-US' === '01:00pm'
- [x] Test 'es-ES' === '13:00'

# Video of testing the Switch and deletion of localStorage
https://www.loom.com/share/7ca400a8cf1e438b9841b3eac98c39b7

## Checklist:
